README.md

```diff
@@ -86,83 +85,68 @@ language.
+which takes care of broadcasting and ordering the transactions across all 
+replicas before feeding them back to the application's *state*. 
 
 Babble is an ordering system that plugs into any application thanks to a very 
-simple JSON-RPC interface over TCP. It uses a consensus algorithm, to replicate 
-and order the transactions, and a blockchain to represent the resulting list. 
-A blockchain is a linear data structure composed of batches of transactions, 
-hashed and signed together, easily allowing to verify any transaction. So, 
-instead of applying commands directly to the *state*, Babble applications must 
-forward the commands to Babble and let them be processed asynchronously by the 
-consensus system before receiving them back, in blocks, ready to be applied 
-to the *state*.  
+simple interface. It uses a consensus algorithm, to replicate and order the 
+transactions, and a blockchain to represent the resulting list. A blockchain is 
+a linear data structure composed of batches of transactions, hashed and signed 
+together, easily allowing to verify any transaction. So, instead of applying 
+commands directly to the *state*, Babble applications must forward the commands 
+to Babble and let them be processed asynchronously by the consensus system 
+before receiving them back, in blocks, ready to be applied to the *state*.  
 
-## Build from source
+### API
 
-The easiest way to build binaries is to do so in a hermetic Docker container. 
-Use this simple command:
+Babble communicates with the App through an `AppProxy` interface, which has two
+implementations:
 
-```bash
-[...]/babble$ make dist
-```
-This will launch the build in a Docker container and write all the artifacts in
-the build/ folder.
+- `SocketProxy`: A SocketProxy connects to an App via TCP sockets. It enables 
+                 the application to run in a separate process or machine, and to 
+                 be written in any programming language.
 
-```bash
-[...]/babble$ tree build
-build/
-├── dist
-│   ├── babble_0.1.0_darwin_386.zip
-│   ├── babble_0.1.0_darwin_amd64.zip
-│   ├── babble_0.1.0_freebsd_386.zip
-│   ├── babble_0.1.0_freebsd_arm.zip
-│   ├── babble_0.1.0_linux_386.zip
-│   ├── babble_0.1.0_linux_amd64.zip
-│   ├── babble_0.1.0_linux_arm.zip
-│   ├── babble_0.1.0_SHA256SUMS
-│   ├── babble_0.1.0_windows_386.zip
-│   └── babble_0.1.0_windows_amd64.zip
-└── pkg
-    ├── darwin_386
-    │   └── babble
-    ├── darwin_amd64
-    │   └── babble
-    ├── freebsd_386
-    │   └── babble
-    ├── freebsd_arm
-    │   └── babble
-    ├── linux_386
-    │   └── babble
-    ├── linux_amd64
-    │   └── babble
-    ├── linux_arm
-    │   └── babble
-    ├── windows_386
-    │   └── babble.exe
-    └── windows_amd64
-        └── babble.exe
-```
+- `InmemProxy` : An InmemProxy uses native callback handlers to integrate Babble 
+                 as a regular Go dependency. 
+
+The `AppProxy` interface exposes three methods for Babble to call the App:
```

Comment:
Here I would have explain the BabbleProxy instead,  as the user never need to be aware of the details of implementation of that part of the proxy. Maybe also add the details of the ProxyHandler interface ?